examples/vision/mixUp.py

```diff
@@ -0,0 +1,218 @@
+"""
+Title: mixup augmentation for image classification
+Author: [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2021/03/06
+Last modified: 2021/03/06
+Description: Data augmentation using the mixup technique for image classification.
+"""
+"""
+## Introduction
+"""
+
+"""
+mixup is a *domain-agnostic* data augmentation technique proposed in [mixup: Beyond
+Empirical Risk Minimization](https://arxiv.org/abs/1710.09412) by Zhang et al. It's
+implemented with the following formulas:
+
+* $\tilde{x}=\lambda x_{i}+(1-\lambda) x_{j}$, where $x_{i}$ and $x_{j}$ are input
+features
+* $\bar{y}=\lambda y_{i}+(1-\lambda) y_{j}$, where $y_{i}$ and $y_{j}$ are one-hot
+encoded labels
+
+(Note that $lambda x_{i}$ and $lambda y_{i}$ are values with the [0, 1] range and are
+sampled from the [Beta distribution](https://en.wikipedia.org/wiki/Beta_distribution).)
+
+The technique is quite holistically named - we are literally mixing up the features and
+their corresponding labels. Implementation-wise it's simple. mixup can be extended to a
+variety of data modalities such as computer vision, natural language processing, speech,
+and so on.
+
+This example requires TensorFlow 2.4 or higher, as well as TensorFlow Probability, which
+can be installed using the following command:
+"""
+
+"""shell
+pip install tensorflow-probability
+"""
+
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+import tensorflow.keras.layers as L
+from tensorflow.keras.utils import to_categorical
+
+import tensorflow_probability as tfp
+
+tfd = tfp.distributions
+
+"""
+## Prepare the dataset
+
+In this example, we will be using the [FashionMNIST](https://research.zalando.com/welcome/mission/research-projects/fashion-mnist/) dataset. But this same recipe can
+be used for other classification datasets as well.
+"""
+
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.fashion_mnist.load_data()
```

Comment:
Given that the technique is domain-specific, would it be more interesting to test it on a text dataset, rather than an image dataset for which other augmentation techniques are available?

Reply:
For an introduction, I think an image dataset is fine.